### PR TITLE
add content on generating certificates

### LINKED ISSUE #88: Update Besu/Tessera TLS info
See https://github.com/hyperledger/besu/issues/2045

### PR DESCRIPTION
**Pull request by rolandtyler**
## Pull Request Description
Added content on generating certificates for Tessera.  
Minor updates to TLS page.

## Fixed Issue(s)

Fixes #88 

## Preview
https://pegasys-tessera--112.com.readthedocs.build/en/112/

### CODE REVIEW

**Review by macfarla — APPROVED** [review]
LGTM

File: docs/HowTo/Generate-certificates.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+These procedures explain how to use [OpenSSL](https://www.openssl.org/source/) to generate certificates
+when the Common Name (CN) is either the [public DNS](#public-dns-as-cn) or an [IP address](#ip-address-as-cn).
+Before you begin, ensure OpenSSL is installed.
+
+### Public DNS as CN
+
+To use a public DNS as CN:
+
+#### Generating a CA certificate
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:20]:
```suggestion
This guide explains how to use [OpenSSL](https://www.openssl.org/source/) to generate certificates
when the Common Name (CN) is either the [public DNS](#public-dns-as-cn) or an [IP address](#ip-address-as-cn).
Before you begin, ensure OpenSSL is installed.

## Public DNS as CN

Follow these steps to use a public DNS as CN.

### Generating a CA certificate
```

I think you can remove one level of headings.

File: docs/HowTo/Generate-certificates.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+1. Generate a key file called `tessera_ca.key`:
+
+    ```bash
+    openssl genrsa -out tessera_ca.key 2048
+    ```
+
+2. Generate a certificate authority [(CA)](Configure/TLS.md#ca) certificate called `tessera_ca.pem` that uses `tessera_ca.key`:
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:28]:
```suggestion
2. Generate a certificate authority (CA) certificate called `tessera_ca.pem` that uses `tessera_ca.key`:
```

File: docs/HowTo/Generate-certificates.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    ```bash
+    openssl req -x509 -new -nodes -key tessera_ca.key -sha256 -days 1024 -out tessera_ca.pem
+    ```
+
+#### Generating a new certificate for a node
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:34]:
```suggestion
### Generating a new certificate for a node
```

File: docs/HowTo/Generate-certificates.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    ```bash
+    openssl x509 -req -in tessera_cer.csr -CA tessera_ca.pem -CAkey tessera_ca.key -CAcreateserial -out tessera_cer.pem -days 500 -sha256
+    ```
+
+### IP address as CN
+
+To use a public IP address as CN:
+
+#### Updating the `openssl.cnf` file
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:67]:
```suggestion
## IP address as CN

Follow these steps to use a public IP address as CN.

### Updating the `openssl.cnf` file
```

File: docs/HowTo/Generate-certificates.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+3. For each DNS you want to use as an alternate name, specify a DNS._n_ entry.
+
+4. For each IP address you want as an alternate IP address, specify an IP._n_ entry.
+
+    !!! note
+
+        When running on `localhost`, make sure to include `127.0.0.1` as a listed IP address.
+
+#### Generating a new CSR for a node
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:96]:
```suggestion
3. For each DNS you want to use as an alternate name, specify a `DNS.n` entry.

4. For each IP address you want as an alternate IP address, specify an `IP.n` entry.

    !!! note

        When running on `localhost`, include `127.0.0.1` as a listed IP address.

### Generating a new CSR for a node
```

File: docs/HowTo/Generate-certificates.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        ```bash
+        [...]
+        Requested Extensions:
+            X509v3 Subject Alternative Name:
+        DNS:<DNS-PUBLIC-RECORD>,
+        DNS:<DNS-PRIVATE-RECORD>,
+        IP Address:<PUBLIC-IP-ADDRESS>,
+        IP Address:<PRIVATE-IP-ADDRESS>
+        [...]
+        ```
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:121]:
```suggestion
    === "Command"
    
        ```bash
        openssl req -text -noout -in tessera_cer.csr
        ```

    === "Output example"

        ```bash
        [...]
        Requested Extensions:
            X509v3 Subject Alternative Name:
        DNS:<DNS-PUBLIC-RECORD>,
        DNS:<DNS-PRIVATE-RECORD>,
        IP Address:<PUBLIC-IP-ADDRESS>,
        IP Address:<PRIVATE-IP-ADDRESS>
        [...]
        ```
```

File: docs/HowTo/Generate-certificates.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+#### Generating a new certificate
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:123]:
```suggestion
### Generating a new certificate
```

File: docs/HowTo/Generate-certificates.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    openssl x509 -req -in tessera_cer.csr -CA tessera_ca.pem -CAkey tessera_ca.key -CAcreateserial -out tessera_cer.pem -days 500 -sha256 -extfile <PATH-TO>/openssl.cnf -extensions v3_req
+    ```
+
+2. Test whether the generated certificate contains the subject alternative names:
+
+    ```bash
+    openssl x509 -in tessera_cer.pem -text -noout
+    ```
+
+    !!! Example "Example of command output"
```
Comment (alexandratran) [inline docs/HowTo/Generate-certificates.md:137]:
```suggestion
    === "Command"
    
        ```bash
        openssl x509 -in tessera_cer.pem -text -noout
        ```

    === "Output example"
```

**Review by alexandratran — COMMENTED** [review]
Added some suggestions.

**Review by alexandratran — APPROVED** [review]

**Pull request merged**